File: Go/1493. Longest Subarray of 1's After Deleting One Element.go
package main

//goland:noinspection GoUnusedFunction
func longestSubarray(nums []int) int {
	left := 0
	right := 0

	for remove := 1; right < len(nums); right++ {
		if nums[right] == 0 {
			remove--
		}

		if remove < 0 {
			if nums[left] == 0 {
				remove++
			}
			left++
		}
	}

	return right - left - 1
}

File: Python/1493. Longest Subarray of 1's After Deleting One Element.py
from typing import List


# noinspection PyMethodMayBeStatic
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        left = 0
        right = 0
        remove = 1

        for right in range(len(nums)):
            if nums[right] == 0:
                remove -= 1

            if remove < 0:
                if nums[left] == 0:
                    remove += 1
                left += 1

        return right - left
